### PR TITLE
updating links from fwlink to real link if exists - contributes to #3391

### PR DESCRIPTION
**Pull request by eddynaka**
updating links from fwlink to real link if exists - contributes to #3391

### CODE REVIEW

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @eddynaka. Submitting the first comments re. this PR. GitHub tends to crash if I go over 50.

File: docs/framework/data/adonet/sql/table-valued-parameters.md [hunk 1]
```diff
@@ -18,8 +18,8 @@ Table-valued parameters provide an easy way to marshal multiple rows of data fro
   
 |Resource|Description|  
 |--------------|-----------------|  
-|[Table-Valued Parameters (Database Engine)](https://go.microsoft.com/fwlink/?LinkId=98363) in SQL Server Books Online|Describes how to create and use table-valued parameters.|  
-|[User-Defined Table Types](https://go.microsoft.com/fwlink/?LinkId=98364) in SQL Server Books Online|Describes user-defined table types that are used to declare table-valued parameters.|  
+|[Table-Valued Parameters (Database Engine)](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/bb510489(v=sql.100)) in SQL Server Books Online|Describes how to create and use table-valued parameters.|
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/table-valued-parameters.md]:
```suggestion
|[Use Table-Valued Parameters (Database Engine)](/sql/relational-databases/tables/use-table-valued-parameters-database-engine)|Describes how to create and use table-valued parameters.|  
```

File: docs/framework/data/adonet/sql/table-valued-parameters.md [hunk 2]
```diff
@@ -18,8 +18,8 @@ Table-valued parameters provide an easy way to marshal multiple rows of data fro
   
 |Resource|Description|  
 |--------------|-----------------|  
-|[Table-Valued Parameters (Database Engine)](https://go.microsoft.com/fwlink/?LinkId=98363) in SQL Server Books Online|Describes how to create and use table-valued parameters.|  
-|[User-Defined Table Types](https://go.microsoft.com/fwlink/?LinkId=98364) in SQL Server Books Online|Describes user-defined table types that are used to declare table-valued parameters.|  
+|[Table-Valued Parameters (Database Engine)](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/bb510489(v=sql.100)) in SQL Server Books Online|Describes how to create and use table-valued parameters.|  
+|[User-Defined Table Types](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/bb522526(v=sql.100)) in SQL Server Books Online|Describes user-defined table types that are used to declare table-valued parameters.|
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/table-valued-parameters.md]:
```suggestion
|[User-Defined Table Types](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/bb522526(v=sql.100))|Describes user-defined table types that are used to declare table-valued parameters.|  
```

File: docs/framework/data/adonet/sql/table-valued-parameters.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ Table-valued parameters provide an easy way to marshal multiple rows of data fro
...
   
 ## Creating Table-Valued Parameter Types  
- Table-valued parameters are based on strongly-typed table structures that are defined by using Transact-SQL CREATE TYPE statements. You have to create a table type and define the structure in SQL Server before you can use table-valued parameters in your client applications. For more information about creating table types, see [User-Defined Table Types](https://go.microsoft.com/fwlink/?LinkID=98364) in SQL Server Books Online.  
+ Table-valued parameters are based on strongly-typed table structures that are defined by using Transact-SQL CREATE TYPE statements. You have to create a table type and define the structure in SQL Server before you can use table-valued parameters in your client applications. For more information about creating table types, see [User-Defined Table Types](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/bb522526(v=sql.100)) in SQL Server Books Online.
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/table-valued-parameters.md]:
```suggestion
 Table-valued parameters are based on strongly-typed table structures that are defined by using Transact-SQL CREATE TYPE statements. You have to create a table type and define the structure in SQL Server before you can use table-valued parameters in your client applications. For more information about creating table types, see [User-Defined Table Types](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/bb522526(v=sql.100)).  
```

File: docs/framework/data/adonet/sql/the-context-connection.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ The problem of internal data access is a fairly common scenario. That is, you wi
   
  **SQL Server Books Online**
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/the-context-connection.md]:
```suggestion
 **SQL Server documentation**  
```

File: docs/framework/data/adonet/sql/sql-server-common-language-runtime-integration.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ SQL Server 2005 introduced the integration of the common language runtime (CLR)
   
  **SQL Server Books Online**  
   
-1. [Common Language Runtime (CLR) Integration Programming Concepts](https://go.microsoft.com/fwlink/?LinkId=115240)  
+1. [Common Language Runtime (CLR) Integration Programming Concepts](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/ms131102(v=sql.100))
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/sql-server-common-language-runtime-integration.md]:
```suggestion
1. [Common Language Runtime (CLR) Integration Programming Concepts](/sql/relational-databases/clr-integration/common-language-runtime-clr-integration-programming-concepts)  
```

File: docs/framework/data/adonet/sql/sql-server-data-types.md [hunk 6]
```diff
@@ -15,7 +15,7 @@ SQL Server and the .NET Framework are based on different type systems, which can
   
  **SQL Server Books Online**  
   
-1. [Data Types (Database Engine)](https://go.microsoft.com/fwlink/?LinkID=107468)  
+1. [Data Types (Database Engine)](https://docs.microsoft.com/previous-versions/sql/sql-server-2008-r2/ms187594(v=sql.105))
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/sql-server-data-types.md]:
```suggestion
1. [Data Types (Transact-SQL)](/sql/t-sql/data-types/data-types-transact-sql)  
```

File: docs/framework/data/adonet/sql/sql-server-in-process-specific-behavior-of-adonet.md [hunk 7]
```diff
@@ -11,7 +11,7 @@ There are four main functional extensions to ADO.NET, found in the <xref:Microso
   
  **SQL Server Books Online**  
   
-1. [SQL Server In-Process Specific Extensions to ADO.NET](https://go.microsoft.com/fwlink/?LinkId=115403)  
+1. [SQL Server In-Process Specific Extensions to ADO.NET](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/ms131070(v=sql.100))
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/sql-server-in-process-specific-behavior-of-adonet.md]:
```suggestion
1. [SQL Server In-Process Specific Extensions to ADO.NET](/sql/relational-databases/clr-integration-data-access-in-process-ado-net/sql-server-in-process-specific-extensions-to-ado-net)  
```

File: docs/framework/data/adonet/sql/the-context-connection.md [hunk 8]
```diff
@@ -10,7 +10,7 @@ The problem of internal data access is a fairly common scenario. That is, you wi
   
  **SQL Server Books Online**  
   
-1. [The Context Connection](https://go.microsoft.com/fwlink/?LinkId=115395)  
+1. [The Context Connection](https://docs.microsoft.com/previous-versions/sql/sql-server-2008/ms131053(v=sql.100))
```
Comment (mairaw) [inline docs/framework/data/adonet/sql/the-context-connection.md]:
```suggestion
1. [The Context Connection](/sql/relational-databases/clr-integration/data-access/context-connection)  
```

File: docs/framework/data/adonet/sql/date-and-time-data.md [hunk 9]
```diff
@@ -229,9 +229,9 @@ command.Parameters.AddWithValue( _
```
[the remark this answers is not in the document]
Comment (eddynaka) [reply docs/framework/data/adonet/sql/date-and-time-data.md]: can you check if thats what you wanted, pls? Thanks

**Comment by mairaw** [comment]
We were having some build issues yesterday

File: docs/framework/data/adonet/sql/date-and-time-data.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (mairaw) [reply docs/framework/data/adonet/sql/date-and-time-data.md]: yes, thanks!

**Review by mairaw — APPROVED** [review]
Thank you @eddynaka.

**Pull request merged**